src/metric.c

```diff
@@ -182,6 +182,13 @@ histogram__sample(struct brubeck_metric *metric, brubeck_sample_cb sample, void
 		sample(key, hsample.count, opaque);
 	}
 
+	if (opaque != NULL){
```

Comment:
`opaque` is assured to be non-NULL, so you can skip this check and move the cast inside the `WITH_SUFFIX` block. :)